### ISSUE TITLE
Goemesa HBase Tutorial: Call failed on IOException

### ISSUE THREAD

**Issue by xlwang233** [geomesa/geomesa-tutorials#67]
Hello,

I followed instructions https://www.geomesa.org/documentation/tutorials/geomesa-quickstart-hbase.html
I installed geomesa-hbase_2.11-2.0.2 and set up HBase in standalone mode (HBase version is 1.3.1 ) on one of my virtual machine (CentOS 6.9).  When I ran the tutorial on the command line nothing wrong happened until at the end I got a "Call failed on IOExceptoin". It seems that the jar has been successfully run and all the intermediate results has been calculated but it failed to output the result to the HBase table. The detailed logs are showned below:
```
Error calling coprocessor service org.locationtech.geomesa.hbase.proto.GeoMesaProto$GeoMesaCoprocessorService for row 
java.util.concurrent.ExecutionException: java.io.IOException: java.io.InterruptedIOException: Interrupted after 5 tries  on 35
        at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:185)
        at org.apache.hadoop.hbase.ipc.RegionCoprocessorRpcChannel.callExecService(RegionCoprocessorRpcChannel.java:103)
        at org.apache.hadoop.hbase.ipc.CoprocessorRpcChannel.callMethod(CoprocessorRpcChannel.java:56)
        at org.locationtech.geomesa.hbase.proto.GeoMesaProto$GeoMesaCoprocessorService$Stub.getResult(GeoMesaProto.java:1146)
        at org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor$$anon$2$$anon$3.call(GeoMesaCoprocessor.scala:133)
        at org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor$$anon$2$$anon$3.call(GeoMesaCoprocessor.scala:129)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
        at org.apache.hadoop.hbase.client.HTable.coprocessorService(HTable.java:1775)
        at org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor$$anon$1.liftedTree1$1(GeoMesaCoprocessor.scala:163)
        at org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor$$anon$1.result$lzycompute(GeoMesaCoprocessor.scala:163)
        at org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor$$anon$1.result(GeoMesaCoprocessor.scala:161)
        at org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor$$anon$1.hasNext(GeoMesaCoprocessor.scala:170)
        at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:369)
        at org.locationtech.geomesa.utils.collection.CloseableIterator$$anon$2.hasNext(CloseableIterator.scala:35)
        at scala.collection.Iterator$class.foreach(Iterator.scala:742)
        at org.locationtech.geomesa.utils.collection.CloseableIterator$$anon$2.foreach(CloseableIterator.scala:34)
        at scala.collection.generic.Growable$class.$plus$plus$eq(Growable.scala:59)
        at scala.collection.mutable.ListBuffer.$plus$plus$eq(ListBuffer.scala:183)
        at scala.collection.mutable.ListBuffer.$plus$plus$eq(ListBuffer.scala:45)
        at scala.collection.TraversableOnce$class.to(TraversableOnce.scala:308)
        at org.locationtech.geomesa.utils.collection.CloseableIterator$$anon$2.to(CloseableIterator.scala:34)
        at scala.collection.TraversableOnce$class.toList(TraversableOnce.scala:292)
        at org.locationtech.geomesa.utils.collection.CloseableIterator$$anon$2.toList(CloseableIterator.scala:34)
        at org.locationtech.geomesa.hbase.data.HBaseDataStore$$anonfun$4$$anonfun$apply$3$$anonfun$apply$4$$anonfun$apply$5.apply(HBaseDataStore.scala:124)
        at org.locationtech.geomesa.hbase.data.HBaseDataStore$$anonfun$4$$anonfun$apply$3$$anonfun$apply$4$$anonfun$apply$5.apply(HBaseDataStore.scala:123)
        at org.locationtech.geomesa.utils.io.WithClose$.apply(SafeClose.scala:51)
        at org.locationtech.geomesa.hbase.data.HBaseDataStore$$anonfun$4$$anonfun$apply$3$$anonfun$apply$4.apply(HBaseDataStore.scala:123)
        at org.locationtech.geomesa.hbase.data.HBaseDataStore$$anonfun$4$$anonfun$apply$3$$anonfun$apply$4.apply(HBaseDataStore.scala:122)
        at org.locationtech.geomesa.utils.io.WithClose$.apply(SafeClose.scala:51)
        at org.locationtech.geomesa.hbase.data.HBaseDataStore$$anonfun$4$$anonfun$apply$3.apply(HBaseDataStore.scala:122)
        at org.locationtech.geomesa.hbase.data.HBaseDataStore$$anonfun$4$$anonfun$apply$3.apply(HBaseDataStore.scala:117)
        at scala.collection.parallel.AugmentedIterableIterator$class.flatmap2combiner(RemainsIterator.scala:132)
        at scala.collection.parallel.immutable.ParVector$ParVectorIterator.flatmap2combiner(ParVector.scala:62)
        at scala.collection.parallel.ParIterableLike$FlatMap.leaf(ParIterableLike.scala:1072)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply$mcV$sp(Tasks.scala:49)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
        at scala.collection.parallel.Task$class.tryLeaf(Tasks.scala:51)
        at scala.collection.parallel.ParIterableLike$FlatMap.tryLeaf(ParIterableLike.scala:1068)
        at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask$class.compute(Tasks.scala:152)
        at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:443)
        at scala.concurrent.forkjoin.RecursiveAction.exec(RecursiveAction.java:160)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinTask.doJoin(ForkJoinTask.java:341)
        at scala.concurrent.forkjoin.ForkJoinTask.join(ForkJoinTask.java:673)
        at scala.collection.parallel.ForkJoinTasks$WrappedTask$class.sync(Tasks.scala:378)
        at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.sync(Tasks.scala:443)
        at scala.collection.parallel.ForkJoinTasks$class.executeAndWaitResult(Tasks.scala:426)
        at scala.collection.parallel.ForkJoinTaskSupport.executeAndWaitResult(TaskSupport.scala:56)
        at scala.collection.parallel.ExecutionContextTasks$class.executeAndWaitResult(Tasks.scala:558)
        at scala.collection.parallel.ExecutionContextTaskSupport.executeAndWaitResult(TaskSupport.scala:80)
        at scala.collection.parallel.ParIterableLike$ResultMapping.leaf(ParIterableLike.scala:958)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply$mcV$sp(Tasks.scala:49)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
        at scala.collection.parallel.Task$class.tryLeaf(Tasks.scala:51)
        at scala.collection.parallel.ParIterableLike$ResultMapping.tryLeaf(ParIterableLike.scala:953)
        at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask$class.compute(Tasks.scala:152)
        at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:443)
        at scala.concurrent.forkjoin.RecursiveAction.exec(RecursiveAction.java:160)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinTask.doJoin(ForkJoinTask.java:341)
        at scala.concurrent.forkjoin.ForkJoinTask.join(ForkJoinTask.java:673)
        at scala.collection.parallel.ForkJoinTasks$WrappedTask$class.sync(Tasks.scala:378)
        at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.sync(Tasks.scala:443)
        at scala.collection.parallel.ForkJoinTasks$class.executeAndWaitResult(Tasks.scala:426)
        at scala.collection.parallel.ForkJoinTaskSupport.executeAndWaitResult(TaskSupport.scala:56)
        at scala.collection.parallel.ExecutionContextTasks$class.executeAndWaitResult(Tasks.scala:558)
        at scala.collection.parallel.ExecutionContextTaskSupport.executeAndWaitResult(TaskSupport.scala:80)
        at scala.collection.parallel.ParIterableLike$class.flatMap(ParIterableLike.scala:513)
        at scala.collection.parallel.immutable.ParVector.flatMap(ParVector.scala:38)
        at org.locationtech.geomesa.hbase.data.HBaseDataStore$$anonfun$4.apply(HBaseDataStore.scala:117)
        at org.locationtech.geomesa.hbase.data.HBaseDataStore$$anonfun$4.apply(HBaseDataStore.scala:116)
        at scala.collection.parallel.mutable.ParArray$ParArrayIterator.flatmap2combiner(ParArray.scala:417)
        at scala.collection.parallel.ParIterableLike$FlatMap.leaf(ParIterableLike.scala:1072)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply$mcV$sp(Tasks.scala:49)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
        at scala.collection.parallel.Task$class.tryLeaf(Tasks.scala:51)
        at scala.collection.parallel.ParIterableLike$FlatMap.tryLeaf(ParIterableLike.scala:1068)
        at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask$class.compute(Tasks.scala:152)
        at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:443)
        at scala.concurrent.forkjoin.RecursiveAction.exec(RecursiveAction.java:160)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinTask.doJoin(ForkJoinTask.java:341)
        at scala.concurrent.forkjoin.ForkJoinTask.join(ForkJoinTask.java:673)
        at scala.collection.parallel.ForkJoinTasks$WrappedTask$class.sync(Tasks.scala:378)
        at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.sync(Tasks.scala:443)
        at scala.collection.parallel.ForkJoinTasks$class.executeAndWaitResult(Tasks.scala:426)
        at scala.collection.parallel.ForkJoinTaskSupport.executeAndWaitResult(TaskSupport.scala:56)
        at scala.collection.parallel.ExecutionContextTasks$class.executeAndWaitResult(Tasks.scala:558)
        at scala.collection.parallel.ExecutionContextTaskSupport.executeAndWaitResult(TaskSupport.scala:80)
        at scala.collection.parallel.ParIterableLike$ResultMapping.leaf(ParIterableLike.scala:958)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply$mcV$sp(Tasks.scala:49)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
        at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
        at scala.collection.parallel.Task$class.tryLeaf(Tasks.scala:51)
        at scala.collection.parallel.ParIterableLike$ResultMapping.tryLeaf(ParIterableLike.scala:953)
        at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask$class.compute(Tasks.scala:152)
        at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:443)
        at scala.concurrent.forkjoin.RecursiveAction.exec(RecursiveAction.java:160)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.io.IOException: java.io.InterruptedIOException: Interrupted after 5 tries  on 35
        at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:185)
        at org.apache.hadoop.hbase.ipc.RegionCoprocessorRpcChannel.callExecService(RegionCoprocessorRpcChannel.java:103)
        at org.apache.hadoop.hbase.ipc.CoprocessorRpcChannel.callMethod(CoprocessorRpcChannel.java:56)
        at org.locationtech.geomesa.hbase.proto.GeoMesaProto$GeoMesaCoprocessorService$Stub.getResult(GeoMesaProto.java:1146)
        at org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor$$anon$2$$anon$3.call(GeoMesaCoprocessor.scala:133)
        at org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor$$anon$2$$anon$3.call(GeoMesaCoprocessor.scala:129)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

        at org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor$$anon$2.call(GeoMesaCoprocessor.scala:152)
        at org.locationtech.geomesa.hbase.coprocessor.GeoMesaCoprocessor$$anon$2.call(GeoMesaCoprocessor.scala:125)
        at org.apache.hadoop.hbase.client.HTable$15.call(HTable.java:1763)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

I checked my {hbase.root.dir}/data/default. A table has been created with name specified as when running the jar. But the table only has 7 rows which look like this:

```
hbase(main):003:0> scan 'hbasetutorialquickstart'
*********************************************************************************************************
ROW                     COLUMN+CELL                                                      
 gdelt-quickstart~attri column=m:v, timestamp=1533132944618, value=GLOBALEVENTID:String,A
 butes                  ctor1Name:String,Actor1CountryCode:String,Actor2Name:String,Actor
                        2CountryCode:String,EventCode:String:index=true,NumMentions:Integ
                        er,NumSources:Integer,NumArticles:Integer,ActionGeo_Type:Integer,
                        ActionGeo_FullName:String,ActionGeo_CountryCode:String,dtg:Date,*
                        geom:Point:srid=4326;geomesa.index.dtg='dtg',geomesa.table.sharin
                        g='false',geomesa.indices='z3:2:3,z2:2:3,id:1:3,attr:5:3'        
 gdelt-quickstart~id    column=m:v, timestamp=1533132944618, value=\x01                  
 gdelt-quickstart~stats column=m:v, timestamp=1533132944618, value=2018-08-01T14:15:43.06
 -date                  0Z                                                               
 gdelt-quickstart~table column=m:v, timestamp=1533132948575, value=hbasetutorialquickstar
 .attr.v5               t_gdelt_2dquickstart_attr_v5                                     
 gdelt-quickstart~table column=m:v, timestamp=1533132947292, value=hbasetutorialquickstar
 .id.v1                 t_gdelt_2dquickstart_id                                          
 gdelt-quickstart~table column=m:v, timestamp=1533132946049, value=hbasetutorialquickstar
 .z2.v2                 t_gdelt_2dquickstart_z2_v2                                       
 gdelt-quickstart~table column=m:v, timestamp=1533132944742, value=hbasetutorialquickstar
 .z3.v2                 t_gdelt_2dquickstart_z3_v2                                       
7 row(s) in 0.3070 seconds
*********************************************************************************************************
```

I am new to HBase and Geomesa and I cannot google out any solution. That's why I come here for help.

Any help is welcomed!  Many thanks in advance!

Xinglei

**Comment by xlwang233** [comment]
OMG. I don't know what's wrong with the format, it's such a mess. I am so sorry I am new to github,too. 

Please read this issue! Grateful for your patience!

**Comment by elahrvivaz** [comment]
did you follow the directions [here](https://www.geomesa.org/documentation/user/hbase/install.html#deploying-the-geomesa-hbase-distributed-runtime-jar) on installing the geomesa distributed runtime jar, and registering the coprocessors?
(I formatted your original post - FYI you can use triple back-ticks (`) around a block of text to render it as unformatted code)

**Comment by xlwang233** [comment]
Thank you elahrvivaz. I did install the geomesa distributed runtime jar by copy it to $HBase_HOME/lib and register the coprocessors site-wide using hbase-site.xml. And I have added the directory to the runtime jar to the HBASE_CLASSPATH property. I used the geomesa command line tool "geomesa classpath" to verify that I have successfully done that.  

Actually the tutorial can run and it returned the "right" log information just as the result showed in tutorial guide. But after the "Done" message appeared the "Call failed on IOException ".

```...... ......
Running query EventCode = '051' AND dtg DURING 2017-12-31T00:00:00+00:00/2018-01-02T00:00:00+00:00
Returning attributes [GLOBALEVENTID, dtg, geom]
01 719024909=719024909|2018-01-01T00:00:00.000Z|POINT (144.967 -37.8167)
02 719025178=719025178|2018-01-01T00:00:00.000Z|POINT (151.217 -33.8833)
03 719025965=719025965|2018-01-01T00:00:00.000Z|POINT (151.217 -33.8833)
04 719025509=719025509|2018-01-01T00:00:00.000Z|POINT (135 -25)
05 719025742=719025742|2018-01-01T00:00:00.000Z|POINT (-117.157 32.7153)
06 719025745=719025745|2018-01-01T00:00:00.000Z|POINT (-119.746 36.17)
07 719025743=719025743|2018-01-01T00:00:00.000Z|POINT (-97.3375 37.6922)
08 719027205=719027205|2018-01-01T00:00:00.000Z|POINT (-96.4003 42.5)
09 719025111=719025111|2018-01-01T00:00:00.000Z|POINT (-82.7165 34.9168)
10 719026938=719026938|2018-01-01T00:00:00.000Z|POINT (-77 40.1254)

Returned 138 total features

Done
Call failed on IOException
java.io.InterruptedIOException: Interrupted after 5 tries  on 35
        at org.apache.hadoop.hbase.client.RpcRetryingCaller.callWithRetries(RpcRetryingCaller.java:185)
        at org.apache.hadoop.hbase.ipc.RegionCoprocessorRpcChannel.callExecService(RegionCoprocessorRpcChannel.java:103)
        at org.apache.hadoop.hbase.ipc.CoprocessorRpcChannel.callMethod(CoprocessorRpcChannel.java:56)
...... ...... ```
I think there is something wrong with the output process because I got nothing in the result table.

Just for reference the environment is: Hadoop-2.8.4   HBase-1.3.1  Zookeeper-3.4.6 (I ran HBase in a standalone mode and the zookeeper inside it should be 3.4.6)  Geomesa-2.0.2

**Comment by elahrvivaz** [comment]
Did you check out the 2.0.2 tag of the tutorials before running it?

**Comment by elahrvivaz** [comment]
For reference, geomesa stores the actual data in various index tables. The main table just stores the metadata. The index tables will start with the same name as the metadata table, and have a suffix indicating the index type (e.g. `hbasetutorialquickstart_z3_v3`)

**Comment by xlwang233** [comment]
OK I see. Just as you say I mistook metadata table with the index table. I checked the other table and it contains the resulting data, though I still don't know why the exception occurred.
Thanks! 

However I got more problems visulizing the data, which I'll have to leave it to  tomorrow to solve...

**Comment by dhruv-goyal-guavus** [comment]
What is the version of coprocessor jar are you using, This is because there is some inconsistency.
The quickstart example is based on the version 2.1.0, which is not yet released officially.
https://github.com/locationtech/geomesa/releases , here I can only see the version till 2.0.2. There is a file which is present in the latest version, which is required to run this example. You can build hbase-runtime-dist jar to resolve this issue.
Also, this discrepancy  should not happen from Geomesa's end. They should release the example and distributions with the same version

**Comment by xlwang233** [comment]
Thank you for you reply @dhruv-goyal-guavus 
For your question "What is the version of coprocessor jar are you using", the version is 2.0.2.
Actually I tried the quick start as well as other tutorials on my newly created Ubuntu 18.04 virtual machine followed exactly the same steps. Everything went well, no mistake at all. So I guess it's possible that the error occurred because of some incompatibility between geomesa and the OS I used(CentOS 6.9). So just switch to Ubuntu or any other compatible OS would solve the issue.

Besides, I failed visualizing the data **with Leaflet**. The error messages are:
```
xinglei@xinglei-virtual-machine:~/Softwares/geomesa-hbase_2.11-2.0.2$ bin/geomesa-hbase export --output-format leaflet --feature-name gdelt-quickstart --zookeepers localhost --catalog hbasequickstart
ERROR Unable to create output destination /home/xinglei/Softwares/geomesa-hbase_2.11-2.0.2/leaflet/index.html, check permissions.
java.lang.SecurityException: Unable to create output destination /home/xinglei/Softwares/geomesa-hbase_2.11-2.0.2/leaflet/index.html, check permissions.
    at org.locationtech.geomesa.tools.export.formats.LeafletMapExporter$WritableFile$2.ensureWritable(LeafletMapExporter.scala:127)
    at org.locationtech.geomesa.tools.export.formats.LeafletMapExporter$.getDestination(LeafletMapExporter.scala:134)
    at org.locationtech.geomesa.tools.export.formats.LeafletMapExporter.<init>(LeafletMapExporter.scala:55)
    at org.locationtech.geomesa.tools.export.ExportCommand$class.export(ExportCommand.scala:67)
    at org.locationtech.geomesa.hbase.tools.export.HBaseExportCommand.export(HBaseExportCommand.scala:19)
    at org.locationtech.geomesa.tools.export.ExportCommand$$anonfun$execute$1$$anonfun$apply$1.apply(ExportCommand.scala:40)
    at org.locationtech.geomesa.tools.export.ExportCommand$$anonfun$execute$1$$anonfun$apply$1.apply(ExportCommand.scala:40)
    at org.locationtech.geomesa.tools.DataStoreCommand$class.withDataStore(Command.scala:59)
    at org.locationtech.geomesa.hbase.tools.export.HBaseExportCommand.withDataStore(HBaseExportCommand.scala:19)
    at org.locationtech.geomesa.tools.export.ExportCommand$$anonfun$execute$1.apply(ExportCommand.scala:40)
    at org.locationtech.geomesa.tools.export.ExportCommand$$anonfun$execute$1.apply(ExportCommand.scala:40)
    at org.locationtech.geomesa.utils.stats.MethodProfiling$class.profile(MethodProfiling.scala:19)
    at org.locationtech.geomesa.hbase.tools.export.HBaseExportCommand.profile(HBaseExportCommand.scala:19)
    at org.locationtech.geomesa.tools.export.ExportCommand$class.execute(ExportCommand.scala:40)
    at org.locationtech.geomesa.hbase.tools.export.HBaseExportCommand.execute(HBaseExportCommand.scala:19)
    at org.locationtech.geomesa.tools.Runner$class.main(Runner.scala:28)
    at org.locationtech.geomesa.hbase.tools.HBaseRunner$.main(HBaseRunner.scala:21)
    at org.locationtech.geomesa.hbase.tools.HBaseRunner.main(HBaseRunner.scala)
```

The command actually created a new directory named "/leaflet" under the geomesa home directory.
But why it can't create index.html in the output "/leaflet" directory? The owner of the "/leaflet" is xinglei, which is the user that I logged on to so there should not be a problem when I want to create a file in it. I even changed the permission of the "/leaflet" directory (and its parent directory) to "777" but nothing changed. 
Here is the screenshot of the directory: (the directory is created by the "export" command but inside it there is nothing)
![image](https://user-images.githubusercontent.com/35983217/46068447-c1cff580-c1ab-11e8-80e2-b736ab4b61c8.png)

I am sorry I have been busy on other things and have not attend to this issue for quite a while, Hope someone can give me some clue on how to solve this problem.

**Comment by elahrvivaz** [comment]
@xlwang233 you're hitting a bug in the leaflet export: https://geomesa.atlassian.net/browse/GEOMESA-2306
It's been fixed for 2.1, but for 2.0.2 the work-around is to `touch` the output file:
```
touch leaflet/index.html
```

**Comment by elahrvivaz** [comment]
If this is still an issue, please file a bug in JIRA or start a discussion on the user list or gitter. See https://github.com/locationtech/geomesa#join-the-community
Thanks,
